sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClientOptions.cs

```diff
@@ -36,6 +36,11 @@ public enum ServiceVersion
         /// </summary>
         internal ServiceVersion Version { get; }
 
+        /// <summary>
+        /// Used to serialize the given events to JSON.
+        /// </summary>
+        internal ObjectSerializer ObjectSerializer { get; }
```

Comment:
We can probably make this public and have it be get/set. This is something a user may want to set to influence the serialization options. We can add a test to highlight how this works where we set a JsonObjectSerializer into this property after constructing it with some customization options:
e.g. 
```c#
var serializer = new JsonObjectSerializer(
    new JsonSerializerOptions()
    {
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    }); 
```
If we change the properties in our TestClass to use PascalCase, we should see that the Json recorded in our tests uses PascalCase when no naming policy is set, but camelCase when we set it as above.